### PR TITLE
own: add GraphQL resolver for recent view signal.

### PR DESCRIPTION
**Pull request by sashaostrikov**
Test plan:
GraphQL tests updated.

Part of https://github.com/sourcegraph/sourcegraph/issues/51464

### CODE REVIEW

File: enterprise/cmd/frontend/internal/own/resolvers/resolvers.go [hunk 1]
```diff
@@ -181,6 +191,15 @@ func (r *ownResolver) GitBlobOwnership(
...
 
+	viewerResolvers, err := computeRecentViewSignals(ctx, r.logger, r.db, blob.Path(), repoID)
+	if err != nil {
+		return nil, err
+	}
+	for _, resolver := range viewerResolvers {
+		ownerships = append(ownerships, resolver)
+	}
+	total += len(viewerResolvers)
+
```
Comment (cbart) [inline enterprise/cmd/frontend/internal/own/resolvers/resolvers.go:202]: I think with #51637 this just becomes:

```suggestion
  ownerships = append(ownerships, viewerResolvers...)
```

File: enterprise/cmd/frontend/internal/own/resolvers/resolvers.go [hunk 2]
```diff
@@ -379,6 +398,64 @@ func computeRecentContributorSignals(ctx context.Context, db edb.EnterpriseDB, p
...
+				} else {
+					return nil, errors.Wrap(err, "getting user primary email")
+				}
+			}
+			user = userFromDB
+			email = primaryEmail
+			fetchedUsers[userID] = userFromDB
+			userEmails[userID] = primaryEmail
+		}
+		// TODO(sashaostrikov): what to do if email here is empty
```
Comment (cbart) [inline enterprise/cmd/frontend/internal/own/resolvers/resolvers.go:447]: 👀

**Review by cbart — APPROVED** [review]
🚢

File: enterprise/cmd/frontend/internal/own/resolvers/resolvers.go (hunk 2 above)
Comment (cbart) [reply enterprise/cmd/frontend/internal/own/resolvers/resolvers.go:447]: I'm thinking we could add a person resolver constructor that takes a user?

File: enterprise/cmd/frontend/internal/own/resolvers/resolvers.go [hunk 3]
```diff
@@ -86,6 +89,13 @@ func (o *ownershipReasonResolver) ToRecentContributorOwnershipSignal() (graphqlb
 	return nil, false
 }
 
+func (o *ownershipReasonResolver) ToRecentViewOwnershipSignal() (graphqlbackend.RecentViewOwnershipSignalResolver, bool) {
+	if res, ok := o.resolver.(*recentViewOwnershipSignal); ok {
+		return res, true
+	}
+	return nil, false
```
[the remark this answers is not in the document]
Comment (sashaostrikov) [reply enterprise/cmd/frontend/internal/own/resolvers/resolvers.go:96]: exactly this variant nope, but you led me to one thing, I'll make it more concise 👍🏻

File: enterprise/cmd/frontend/internal/own/resolvers/resolvers.go (hunk 2 above)
Comment (sashaostrikov) [reply enterprise/cmd/frontend/internal/own/resolvers/resolvers.go:447]: yep, let me do it as a quick follow-up

**Comment by sourcegraph-buildkite** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.18 kb) | 0.00% (+0.18 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ao/own/recent-view-gql-resolver/compare-report.html;tab=live_object) for a full comparison between the commits 41b2b9c and 083f91d or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Pull request merged**